server/leafnode.go

```diff
@@ -1092,7 +1099,7 @@ func (c *client) updateSmap(sub *subscription, delta int32) {
 	} else {
 		delete(c.leaf.smap, key)
 	}
-	if update && c.flags.isSet(leafAllSubsSent) {
+	if update {
```

Comment:
This change was for a specific condition, not applicable anymore? Was a very specific bug. I think we have a test for it.

Reply:
No because we no longer split/flush in place/release lock/etc..

Reply:
Ah, gothca.